### PR TITLE
hw3 task 1 update

### PR DESCRIPTION
**Pull request by VitaliiSorokin**

### CODE REVIEW

File: hw3/src/Person.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+public class Person {
+    public String name = "Bob";
+    public int age;
```
Comment (vitalii-kliuiev) [inline hw3/src/Person.java]: both should be private

File: hw3/src/MyListImpl.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+    @Override
+    public Object[] toArray() {
+        return new Object[0];
+    }
+
+    @Override
+    public boolean add(Object o) {
+        arr = Arrays.copyOf(arr, arr.length + 1);
+        arr[arr.length - 1] = new Person();
```
Comment (vitalii-kliuiev) [inline hw3/src/MyListImpl.java]: interesting what happens with "o" ?

File: hw3/src/MyListImpl.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        return false;
```
Comment (vitalii-kliuiev) [inline hw3/src/MyListImpl.java]: why "false"

File: hw3/src/MyListImpl.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+
+    @Override
+    public Iterator iterator() {
+        return null;
+    }
+
+    @Override
+    public Object[] toArray() {
+        return new Object[0];
```
Comment (vitalii-kliuiev) [inline hw3/src/MyListImpl.java:35]: return arr

File: hw3/src/MyListImpl.java [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    @Override
+    public boolean isEmpty() {
+        return arr.length == 0;
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        if (o instanceof Person) {
+            for (Person p : arr) {
+                return p.equals(o);
```
Comment (vitalii-kliuiev) [inline hw3/src/MyListImpl.java:21]: you didn't redefine equals in Person, it will not work correctly

File: hw3/src/MyListImpl.java [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+    @Override
+    public Object remove(int index) {
+        return null;
+    }
+
+    @Override
+    public int indexOf(Object o) {
+        for (int i = 0; i < arr.length; i++) {
+            if (o.equals(arr[i])) {
```
Comment (vitalii-kliuiev) [inline hw3/src/MyListImpl.java:102]: the same

File: hw3/src/MyListImpl.java [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+    @Override
+    public void clear() {
+        for (int i = 0; i < arr.length; i++) {
+            arr[i] = null;
+        }
+    }
+
+    @Override
+    public Object get(int index) {
+        return arr[index];
```
Comment (vitalii-kliuiev) [inline hw3/src/MyListImpl.java]: ArrayIndexOutOfBoundsException

File: hw3/src/MyListImpl.java [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+    @Override
+    public boolean addAll(int index, Collection c) {
+        return false;
+    }
+
+    @Override
+    public void clear() {
+        for (int i = 0; i < arr.length; i++) {
+            arr[i] = null;
```
Comment (vitalii-kliuiev) [inline hw3/src/MyListImpl.java:65]: you should be reduce the size of array and cut this element

File: hw3/src/MyListImpl.java [hunk 9]
```diff
@@ -0,0 +1,143 @@
+import java.util.*;
+
+public class MyListImpl implements MyList {
+
+    private Person[] arr = new Person[4];
```
Comment (vitalii-kliuiev) [inline hw3/src/MyListImpl.java:5]: why 4

File: hw3/src/MyListImpl.java [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+
+    @Override
+    public int size() {
+        return arr.length;
```
Comment (vitalii-kliuiev) [inline hw3/src/MyListImpl.java:9]: size 4 with empty array

File: hw3/src/MyListImpl.java [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+
+    @Override
+    public Object get(int index) {
+        return arr[index];
+    }
+
+    @Override
+    public Object set(int index, Object element) {
+        Person prev = arr[index];
+        arr[index] = (Person) element;
```
Comment (vitalii-kliuiev) [inline hw3/src/MyListImpl.java]: ClassCastException

**Review by vitalii-kliuiev — CHANGES_REQUESTED** [review]
Your implementation of List will not work correctly

File: hw3/src/Person.java (hunk 1 above)
Comment (VitaliiSorokin) [reply hw3/src/Person.java]: ok

File: hw3/src/MyListImpl.java (hunk 2 above)
Comment (VitaliiSorokin) [reply hw3/src/MyListImpl.java]: "o" is a redundant argument=)

File: hw3/src/MyListImpl.java (hunk 3 above)
Comment (VitaliiSorokin) [reply hw3/src/MyListImpl.java]: why not?  why the "add" method requires Boolean output?

File: hw3/src/MyListImpl.java (hunk 4 above)
Comment (VitaliiSorokin) [reply hw3/src/MyListImpl.java:35]: I didn't implement this method, that's default Intellij IDEA implementation

File: hw3/src/MyListImpl.java (hunk 7 above)
Comment (VitaliiSorokin) [reply hw3/src/MyListImpl.java]: got it

File: hw3/src/MyListImpl.java (hunk 9 above)
Comment (VitaliiSorokin) [reply hw3/src/MyListImpl.java:5]: my favorite number)

File: hw3/src/MyListImpl.java (hunk 10 above)
Comment (VitaliiSorokin) [reply hw3/src/MyListImpl.java:9]: yes, it is

File: hw3/src/MyListImpl.java (hunk 11 above)
Comment (VitaliiSorokin) [reply hw3/src/MyListImpl.java]: got it

**Review by vitalii-kliuiev — APPROVED** [review]
Good job!

**Pull request merged**